arch/arc/core/arc_smp.c

```diff
@@ -114,7 +114,7 @@ void z_arc_slave_start(int cpu_num)
 
 #ifdef CONFIG_SMP
 
-static void sched_ipi_handler(const void *unused)
+static void sched_ipi_handler(const struct device *unused)
```

Comment:
not following how this ends up a struct device, it is unrelated to devices...

Reply:
Looks like I need to improve the cocci rule: if the parameter does not get casted to struct device into the function, it should not match.